src/main/java/org/tikv/common/operation/iterator/RawScanIterator.java

```diff
@@ -38,23 +38,22 @@ public RawScanIterator(
   }
 
   TiRegion loadCurrentRegionToCache() throws Exception {
-    TiRegion region;
-    try (RegionStoreClient client = builder.build(startKey)) {
-      region = client.getRegion();
-      BackOffer backOffer = ConcreteBackOffer.newScannerNextMaxBackOff();
-      if (limit <= 0) {
-        currentCache = null;
-      } else {
-        while (true) {
+    BackOffer backOffer = ConcreteBackOffer.newScannerNextMaxBackOff();
+    while (true) {
+      try (RegionStoreClient client = builder.build(startKey)) {
```

Comment:
Is this to fix during retry we need to re-get client since cache might change?

Reply:
Yes. As @birdstorm stated in the description.